src/lib/mdns/Advertiser.h

```diff
@@ -285,30 +285,68 @@ class CommissionAdvertisingParameters : public BaseAdvertisingParams<CommissionA
     bool mPairingInstrHasValue = false;
 };
 
-/// Handles advertising of CHIP nodes
+/**
+ * Interface for advertising CHIP DNS-SD services.
+ *
+ * A user of this interface must first initialize the advertiser using the `Init` method.
+ *
+ * Then, whenever advertised services need to be refreshed, the following sequence of events must
+ * occur:
+ * 1. Call the `RemoveServices` method.
+ * 2. Call one of the `Advertise` methods for each service to be added or updated.
+ * 3. Call the `CompleteServiceUpdate` method to complete the update and apply all pending changes.
+ */
 class ServiceAdvertiser
 {
 public:
     virtual ~ServiceAdvertiser() {}
 
-    /// Starts the advertiser. Items 'Advertised' will become visible.
-    /// Must be called before Advertise() calls.
-    virtual CHIP_ERROR Start(chip::Inet::InetLayer * inetLayer, uint16_t port) = 0;
-
-    /// Advertises the CHIP node as an operational node
+    /**
+     * Initializes the advertiser.
+     *
+     * The method must be called before other methods of this class.
+     * If the advertiser has already been initialized, the method exits immediately with no error.
+     */
+    virtual CHIP_ERROR Init(chip::Inet::InetLayer * inetLayer, uint16_t port) = 0;
```

Comment:
What is the `port ` used for here? Why is the port not part of `OperationalAdvertisingParameters` / `CommisionAdvertiseParameters` ?

Reply:
Anyway, I removed that argument.